src/components/Tooltip.tsx

```diff
@@ -0,0 +1,121 @@
+import React, { ReactNode, CSSProperties } from 'react'
+import Portal from './Portal'
+import { usePopperDefault, TOOLTIP_OFFSET } from 'hooks/usePopper'
+import { State, Placement } from '@popperjs/core'
+import styled from 'styled-components'
+import { isElement, isFragment } from 'react-is'
+
+// visibility necessary for correct boundingRect calculation by popper
+const TooltipOuter = styled.div<Pick<TooltipBaseProps, 'isShown'>>`
+  visibility: ${(props): 'hidden' | false => !props.isShown && 'hidden'};
+`
+// can style anything but TOOLTIP_OFFSET fields, position and transform: rotate
+const TooltipArrow = styled.div`
```

Comment:
why all this styling? I thought the library handled styling more gracefully via props? or am i wrong

Reply:
It handle styling via attributes, like vanilla JS would. And even then only positioning through transforms, not actual style styling